--- /dev/null
+++ b/src/proto_audit/bicc.rs
@@ -0,0 +1,13 @@
+//! Bicc header — 245 bytes.
+
+/// Bicc header — 245 bytes.
+pub struct BiccHeader {
+    /// CIC: 18.
+    pub bicc_cic: u32,
+    /// Message Type: Initial address (1).
+    pub isup_message_type: u8,
+    /// Pointer to start of optional part: 6.
+    pub isup_optional_parameter_part_pointer: u8,
+    /// End of optional parameters (0).
+    pub isup_parameter_type: u8,
+}
