--- /dev/null
+++ b/src/proto_audit/ieee802154.rs
@@ -0,0 +1,17 @@
+//! IEEE802154 header — 296 bytes.
+
+/// IEEE802154 header — 296 bytes.
+pub struct IEEE802154Header {
+    /// Frame Control Field: 0xa861, Frame Type: Data, Acknowledge Request, PAN ID Compression, Destination Addressing Mode: Short/16-bit, Frame Version: IEEE Std 802.15.4-2015, Source Addressing Mode: Short/16-bit.
+    pub wpan_fcf: u16,
+    /// Sequence Number: 91.
+    pub wpan_seq_no: u8,
+    /// Destination PAN: 0xdcba.
+    pub wpan_dst_pan: u16,
+    /// Destination: 0x0000.
+    pub wpan_dst16: u16,
+    /// Source: 0x0001.
+    pub wpan_src16: u16,
+    /// FCS: 0x43f1 (Correct).
+    pub wpan_fcs: u16,
+}
