--- /dev/null
+++ b/src/proto_audit/infiniband.rs
@@ -0,0 +1,17 @@
+//! Infiniband header — 290 bytes.
+
+/// Infiniband header — 290 bytes.
+pub struct InfinibandHeader {
+    /// Local Route Header.
+    pub infiniband_lrh: u64,
+    /// Base Transport Header.
+    pub infiniband_bth: u128,
+    /// DETH - Datagram Extended Transport Header.
+    pub infiniband_deth: u64,
+    /// MAD Header - Common Management Datagram.
+    pub infiniband_mad: u128,
+    /// Invariant CRC: 0x1065ef88.
+    pub infiniband_invariant_crc: u32,
+    /// Variant CRC: 0x0037.
+    pub infiniband_variant_crc: u16,
+}
