--- /dev/null
+++ b/src/proto_audit/lon.rs
@@ -0,0 +1,15 @@
+//! Lon header — 12 bytes.
+
+/// Lon header — 12 bytes.
+pub struct LonHeader {
+    /// PPDU: 0x01.
+    pub lon_ppdu: u8,
+    /// NPDU: 9, PDU format: TPDU, Address format: Unicast (2a)/Multicast (2b), Domain length: 8 bit.
+    pub lon_npdu: u8,
+    /// Domain: 01.
+    pub lon_domain: u8,
+    /// TPDU: 0x03, TPDU type: ACKD.
+    pub lon_tpdu: u8,
+    /// Network Variable: 0x810d.
+    pub lon_nv: u16,
+}
