--- /dev/null
+++ b/src/proto_audit/mailslot.rs
@@ -0,0 +1,15 @@
+//! Mailslot header — 25 bytes.
+
+/// Mailslot header — 25 bytes.
+pub struct MailslotHeader {
+    /// Opcode: Write Mail Slot (1).
+    pub mailslot_opcode: u16,
+    /// Priority: 1.
+    pub mailslot_priority: u16,
+    /// Class: Unreliable & Broadcast (2).
+    pub mailslot_class: u16,
+    /// Size: 60.
+    pub mailslot_size: u16,
+    /// Mailslot Name: \MAILSLOT\BROWSE.
+    pub mailslot_name: u128,
+}
