--- /dev/null
+++ b/src/proto_audit/netlink_netfilter.rs
@@ -0,0 +1,13 @@
+//! Netlink-netfilter header — 28 bytes.
+
+/// Netlink-netfilter header — 28 bytes.
+pub struct NetlinknetfilterHeader {
+    /// 0000 0110 .... .... = Subsystem: IP set (6).
+    pub netlink_netfilter_subsys: u16,
+    /// Address family: AF_INET (2).
+    pub netlink_netfilter_family: u8,
+    /// Version: 0.
+    pub netlink_netfilter_version: u8,
+    /// Resource id: 0.
+    pub netlink_netfilter_res_id: u16,
+}
