--- /dev/null
+++ b/src/proto_audit/opensafety_udp.rs
@@ -0,0 +1,17 @@
+//! Opensafety_udp header — 43 bytes.
+
+/// Opensafety_udp header — 43 bytes.
+pub struct Opensafety_udpHeader {
+    /// Transport Version: 0.
+    pub opensafety_udp_transport_version: u8,
+    /// .... ...0 = Data Type: ACyclic.
+    pub opensafety_udp_transport_flags_type: u8,
+    /// Counter: 0x0200 (512).
+    pub opensafety_udp_transport_counter: u16,
+    /// Sender ID: 0x00000053 (83).
+    pub opensafety_udp_transport_sender: u32,
+    /// Datapoint ID: 0x0001 (1).
+    pub opensafety_udp_transport_datapoint: u16,
+    /// Length: 0.
+    pub opensafety_udp_transport_length: u16,
+}
