--- /dev/null
+++ b/src/proto_audit/swils.rs
@@ -0,0 +1,17 @@
+//! Swils header — 48 bytes.
+
+/// Swils header — 48 bytes.
+pub struct SwilsHeader {
+    /// Cmd Code: HLO (0x14).
+    pub swils_opcode: u8,
+    /// Options: 00000000.
+    pub swils_hlo_options: u32,
+    /// Hello Interval (secs): 20.
+    pub swils_hlo_hloint: u32,
+    /// Dead Interval (secs): 80.
+    pub swils_hlo_deadint: u32,
+    /// Recipient Domain ID: 173.
+    pub swils_hlo_rcvdomid: u8,
+    /// Originating Port Idx: 0x00a000.
+    pub swils_hlo_origpidx: u32,
+}
